modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2DeveloperKeysForDXP.testcase

```diff
@@ -0,0 +1,136 @@
+
+		// Needs refactor since it depends on example Avangrid Portals account
+
+		task ("Signs in OKTA to update and go to Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("Select a project and go to activation page") {
+			CustomerPortal.enterOnProject(title = "${testAccount}");
+
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "DXP");
+		}
+
+		task ("Assert that text 'To activate a local Liferay DXP instance, download a development key for your Liferay DXP version' is present") {
+			AssertElementPresent(locator1 = "CP2Utils#CAPITION_ABOVE_MENU");
+		}
+
+		task ("Select 1 key and click in Download key") {
+			CP2ActivationKeys.selectKeyCheckboxByEnvironmentName(environmentName = "Redacted");
+
+			CustomerPortal.clickAtButton(text = "Download Key");
+		}
+
+		task ("Assest the success message") {
+			Alert.viewSuccessMessage();
+		}
+
+		task ("Assert that the Activation Key was downloaded") {
+			CP2FileUtil.assertDownloadedFileIsPresent(downloadedFileName = "activation-key-dxpdevelopment-7.0-nasaice.xml");
+		}
+	}
+
+	@description = "Automation ID: LPS-146194 | Test Summary: Check that the dropdown list contains 5 DXP versions for the user to choose from."
+	@priority = "4"
+	test CanViewDxpVersionsDropDownList {
+		task ("Go to CP2.0 and join a project") {
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+
+			CustomerPortal.enterOnProject(title = "${testAccount}");
+		}
+
+		task ("Go to the product activation page") {
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "${testProduct}");
+		}
+
+		task ("Open the dropdown and verify that there are 5 DXP versions") {
+			CustomerPortal.viewVersions(dropDownItems = "7.4,7.3,7.2,7.1,7.0");
+		}
+	}
+
+	@description = "Automation ID: LPS-146199 | Test Summary: Verify that when clicking on the Deploying Activation Keys article link, the user is redirected to another page."
+	@priority = "3"
+	test ConfirmRedirectionFromDxpToAnotherPage {
+		task ("Go to CP2.0 and join a project") {
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+
+			CustomerPortal.enterOnProject(title = "${testAccount}");
+		}
+
+		task ("Go to the product activation page") {
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "${testProduct}");
+		}
+
+		task ("Click in the link 'deploying activation keys article' and go to the page") {
+			Click(locator1 = "CustomerPortalSite#DEPLOY_ARTICLE_PAGE");
+
+			Navigator.selectSpecificWindow(windowTitle = "Deploying Activation Keys – Liferay Help Center");
+		}
+
+		task ("Assert that a new page is opened and go back to 'Product Activation page'") {
+			AssertElementPresent(locator1 = "CustomerPortalSite#DEPLOY_ACTIVATION_PAGE");
+
+			Navigator.closeWindow();
+		}
+	}
```

Comment:
Hi @styller1 since the tests follow the same steps, would you mind to combine them into only one test?

Since CanViewDxpVersionsDropDownList has only the CustomerPortal.viewVersions(dropDownItems = "7.4,7.3,7.2,7.1,7.0"); assertion, you could add it in the ConfirmRedirectionFromDxpToAnotherPage test and consume less resources from CI.

Reply:
Done